code/src/components/Game.js

```diff
@@ -0,0 +1,112 @@
+import React, { useEffect } from 'react';
+
+import { useDispatch, useSelector } from 'react-redux'
+
+import { rooms } from '../reducers/rooms'
+import { loader } from '../reducers/loader'
+
+import arrow_one from '../assets/arrow-one.svg'
+import arrow_split from '../assets/arrow-split.svg'
+
+export const Game = () => {
+  const username = useSelector(store => store.rooms.username)
+  const room = useSelector(store => store.rooms)
+  // We'll use this variable to check if things are loading or not.
+  const isLoading = useSelector(store => store.loader.isLoading)
+  const dispatch = useDispatch();
+
+  const START_URL = "https://wk16-backend.herokuapp.com/start";
+  const ACTION_URL = "https://wk16-backend.herokuapp.com/action";
+
+  const startGameFetchInfo = {
+    method: 'POST',
+    headers: { 'Content-Type': 'application/json' },
+    body: JSON.stringify({
+      username: username
+    })
+  }
+
+  // This function starts the game, by doing the first fetch from the API.
+  const startGame = () => {
+    dispatch(loader.actions.setLoading(true))
+    // Fetcha, och sedan skicka in responsen (data) till setGameState. Typ.
+    fetch(START_URL, startGameFetchInfo)
+      .then(response => response.json())
+      .then(data => {
+        // Pass "data" (the response from the API containing the coordinates, direction, etc) to the Game State.
+        dispatch(rooms.actions.setGameState(data))
+        dispatch(loader.actions.setLoading(false))
+      })
+  }
+
+  // This function is called on each click of a "Direction" button – i.e. the buttons in each action box with North, East, West or South.
+  const continueGame = (direction) => {
+    dispatch(loader.actions.setLoading(true))
+    fetch(ACTION_URL, {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({
+        username: username,
+        type: "move",
+        direction: direction
+      })
+    })
+      .then(response => response.json())
+      .then(data => {
+        dispatch(rooms.actions.setGameState(data))
+        dispatch(loader.actions.setLoading(false))
+      })
+  }
+
```

Comment:
Interesting to see how you've included the thunks inside the component where you're mapping over each of the actions returned. We understand that you may have done this to help you understand what's happening, but it might be good to create a separate file for the thunks in future projects. This is just some advice as we found it easier to understand the set up for doing the fetch in the thunk and then dispatching the data to the redux state ❤